src/helpers/network.rs

```diff
@@ -55,47 +40,72 @@ impl Debug for ChannelId {
     }
 }
 
-/// Wrapper around a [`PollSender`] to modify the error message to match what the [`NetworkSink`]
-/// requires. The only error that [`PollSender`] will generate is "channel closed", and thus is the
-/// only error message forwarded from this [`NetworkSink`].
-#[pin_project]
-pub struct NetworkSink<T> {
-    #[pin]
-    inner: PollSender<T>,
+pub type MessageChunks = (ChannelId, Vec<u8>);
+
+/// Given any implementation of [`Transport`], a `Network` is able to send and receive
+/// [`MessageChunks`] for a specific query id. The [`Transport`] will receive `NetworkEvents`
+/// containing the `MessageChunks`
+pub struct Network<T> {
+    transport: T,
+    query_id: QueryId,
+    roles: RoleAssignment,
 }
 
-impl<T: Send + 'static> NetworkSink<T> {
-    #[must_use]
-    pub fn new(sender: mpsc::Sender<T>) -> Self {
+impl<T: Transport> Network<T> {
+    pub fn new(transport: T, query_id: QueryId, roles: RoleAssignment) -> Self {
         Self {
-            inner: PollSender::new(sender),
+            transport,
+            query_id,
+            roles,
         }
     }
-}
 
-impl<T: Send + 'static> futures::Sink<T> for NetworkSink<T>
-where
-    Error: From<PollSendError<T>>,
-{
-    type Error = Error;
+    /// sends a [`NetworkEvent`] containing [`MessageChunks`] on the underlying [`Transport`]
+    /// # Errors
+    /// if `message_chunks` fail to be delivered
+    /// # Panics
+    /// if `roles_to_helpers` does not have all 3 roles
+    pub async fn send(&self, message_chunks: MessageChunks) -> Result<(), Error> {
+        let (channel, payload) = message_chunks;
+        let destination = self.roles.identity(channel.role);
```

Comment:
this is the key responsibility of this struct: resolve helper identifiers (something that transport understand) to roles (infra) before sending them up